full-stack-tests/presentation/src/frontend/UnifiedSelection.test.ts

```diff
@@ -185,6 +188,172 @@ describe("Unified Selection", () => {
+              ruleType: RuleTypes.Grouping,
+              class: { schemaName: "BisCore", className: GeometricElement3d.className },
+              groups: [{
+                specType: GroupingSpecificationTypes.Property,
+                propertyName: "CodeValue",
+                createGroupForSingleItem: true,
+              }],
+            }],
+          }],
+        };
+
+        await using<RegisteredRuleset, Promise<void>>(await Presentation.presentation.rulesets().add(groupByPropertyRuleset), async () => {
+          const rootNodes = await Presentation.presentation.getNodes({ imodel: groupingNodesIModel, rulesetOrId: groupByPropertyRuleset.id });
+          Presentation.selection.addToSelection("", groupingNodesIModel, [rootNodes[0].key]);
+          await waitForAllAsyncs([handler, Presentation.selection.getToolSelectionSyncHandler(groupingNodesIModel)!]);
+          expect(groupingNodesIModel.hilited.elements.size).to.be.equal(2);
+          expect(groupingNodesIModel.hilited.elements.hasId(objectIdArray[0])).to.be.true;
+          expect(groupingNodesIModel.hilited.elements.hasId(objectIdArray[1])).to.be.true;
+
+          Presentation.selection.replaceSelection("", groupingNodesIModel, [rootNodes[1].key]);
+          await waitForAllAsyncs([handler, Presentation.selection.getToolSelectionSyncHandler(groupingNodesIModel)!]);
+          expect(groupingNodesIModel.hilited.elements.size).to.be.equal(1);
+          expect(groupingNodesIModel.hilited.elements.hasId(objectIdArray[2])).to.be.true;
+        });
+      });
+
+      it.skip("hilites grouping nodes grouped by label", async () => {
+        const groupByLabelRuleset: Ruleset = {
+          id: "groupByLabelRuleset",
+          rules: [{
+            ruleType: RuleTypes.RootNodes,
+            specifications: [{
+              specType: ChildNodeSpecificationTypes.InstanceNodesOfSpecificClasses,
+              classes: { schemaName: "BisCore", classNames: [GeometricElement3d.className] },
+              arePolymorphic: true,
+              groupByClass: false,
+              groupByLabel: true,
+            }],
+          }],
+        };
+
+        await using<RegisteredRuleset, Promise<void>>(await Presentation.presentation.rulesets().add(groupByLabelRuleset), async () => {
+          const rootNodes = await Presentation.presentation.getNodes({ imodel: groupingNodesIModel, rulesetOrId: groupByLabelRuleset.id });
+          Presentation.selection.addToSelection("", groupingNodesIModel, [rootNodes[0].key]);
+          await waitForAllAsyncs([handler, Presentation.selection.getToolSelectionSyncHandler(groupingNodesIModel)!]);
+          expect(groupingNodesIModel.hilited.elements.size).to.be.equal(2);
+          expect(groupingNodesIModel.hilited.elements.hasId(objectIdArray[0])).to.be.true;
+          expect(groupingNodesIModel.hilited.elements.hasId(objectIdArray[1])).to.be.true;
+
+          Presentation.selection.replaceSelection("", groupingNodesIModel, [rootNodes[1].key]);
```

Comment:
as far as I can tell `rootNodes[1]` is not a label grouping node, so no point to have a test for it here

Reply:
Removed.